tests/testFlagHandler.py

```diff
@@ -166,49 +216,95 @@ def testFlagHandler(self):
+        task.run(cat, exposure)
+        self.assertEqual(source.get(self.algName + "_flag"), True)
+        self.assertEqual(source.get(self.algName + "_flag_containsNan"), True)
+        self.assertEqual(source.get(self.algName + "_flag_edge"), False)
 
-    def testError1(self):
-        self.config.plugins[self.algName].failureType = PythonPlugin.ErrEnum.flag_error1
+    #   Test that the Edge error can be triggered.
+    def testPluginEdgeError(self):
         schema = self.dataset.makeMinimalSchema()
         task = lsst.meas.base.SingleFrameMeasurementTask(schema=schema, config=self.config)
         exposure, cat = self.dataset.realize(noise=100.0, schema=schema)
+        #   Set the size large enough to trigger the edge error
+        self.config.plugins[self.algName].size = exposure.getDimensions()[1]//2
+        task.log.setThreshold(task.log.FATAL)
         task.run(cat, exposure)
         source = cat[0]
         self.assertEqual(source.get(self.algName + "_flag"), True)
-        self.assertEqual(source.get(self.algName + "_flag_error1"), True)
-        self.assertEqual(source.get(self.algName + "_flag_error2"), False)
+        self.assertEqual(source.get(self.algName + "_flag_containsNan"), False)
+        self.assertEqual(source.get(self.algName + "_flag_edge"), True)
 
-    def testError2(self):
-        self.config.plugins[self.algName].failureType = PythonPlugin.ErrEnum.flag_error2
+    #   Test that the SafeCentroidExtractor works and sets the flags correctly
+    def testSafeCentroider(self):
+        #   Normal case should use the centroid slot to get the center, which should succeed
         schema = self.dataset.makeMinimalSchema()
         task = lsst.meas.base.SingleFrameMeasurementTask(schema=schema, config=self.config)
+        task.log.setThreshold(task.log.FATAL)
         exposure, cat = self.dataset.realize(noise=0.0, schema=schema)
-        task.run(cat, exposure)
         source = cat[0]
+        task.run(cat, exposure)
+        self.assertEqual(source.get(self.algName + "_flag"), False)
+        flux = source.get("test_PythonPlugin_flux")
+        self.assertFalse(numpy.isnan(flux))
+
+        #   If one of the center coordinates is nan and the centroid slot error flag has
+        #   not been set, the SafeCentroidExtractor will fail.
+        source.set('truth_x', numpy.nan)
+        source.set('truth_flag', False)
+        source.set("test_PythonPlugin_flux", numpy.nan)
+        source.set(self.algName + "_flag", False)
+        task.run(cat, exposure)
+        self.assertEqual(source.get(self.algName + "_flag"), True)
+        self.assertTrue(numpy.isnan(source.get("test_PythonPlugin_flux")))
+
+        #   But if the same conditions occur and the centroid slot error flag is set
+        #   to true, the SafeCentroidExtractor will succeed and the algorithm will complete.
+        #   However, the failure flag will also be set.
```

Comment:
For this in-line comment and others, there should only be a single space between `#` and the comment text